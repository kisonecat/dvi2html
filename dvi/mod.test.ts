import { color, papersize, parse, ps, svg } from "./mod.ts";
import { assertSnapshot } from "https://deno.land/std@0.210.0/testing/snapshot.ts";

const test = async (t: Deno.TestContext, filename: string) => {
  const result: unknown[] = [];

  for (
    const command of parse(
      await Deno.readFile(new URL(`../test/${filename}`, import.meta.url)),
      [papersize, color(), svg(), ps()],
    )
  ) {
    switch (command.type) {
      case "rect":
      case "info":
      case "color":
      case "papersize":
      case "svg":
        result.push(command);
        break;
      case "special":
        result.push({
          type: command.type,
          data: new TextDecoder().decode(command.data),
        });
        break;
      case "text":
        result.push({
          type: command.type,
          text: command.text,
          font: command.font.name,
          top: command.top,
          left: command.left,
        });
        break;
    }
  }
  await assertSnapshot(t, result);
};

Deno.test("parse()", async (t) => {
  await t.step("color.dvi", () => test(t, "color.dvi"));
  await t.step("main.dvi", () => test(t, "main.dvi"));
});
