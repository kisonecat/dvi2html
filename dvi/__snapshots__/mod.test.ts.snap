export const snapshot = {};

snapshot[`parse() 1`] = `
[
  {
    comment: " TeX output 2023.12.31:1042",
    denominator: 473628672,
    magnification: 1000,
    numerator: 25400000,
    type: "info",
    version: 2,
  },
  {
    data: "header=l3backend-dvips.pro",
    type: "special",
  },
  {
    height: 794.96999,
    type: "papersize",
    width: 614.295,
  },
  {
    color: "black",
    type: "color",
  },
  {
    color: "black",
    type: "color",
  },
  {
    color: "black",
    type: "color",
  },
  {
    font: "cmr10",
    left: 5046272,
    text: "This",
    top: 4128768,
    type: "text",
  },
  {
    font: "cmr10",
    left: 6581570.625,
    text: "example",
    top: 4128768,
    type: "text",
  },
  {
    font: "cmr10",
    left: 9187295.875,
    text: "sho",
    top: 4128768,
    type: "text",
  },
  {
    font: "cmr10",
    left: 10119365.875,
    text: "ws",
    top: 4128768,
    type: "text",
  },
  {
    font: "cmr10",
    left: 11108529.5,
    text: "dieren",
    top: 4128768,
    type: "text",
  },
  {
    font: "cmr10",
    left: 13222072,
    text: "t",
    top: 4128768,
    type: "text",
  },
  {
    font: "cmr10",
    left: 13734278.125,
    text: "examples",
    top: 4128768,
    type: "text",
  },
  {
    font: "cmr10",
    left: 16598506.125,
    text: "on",
    top: 4128768,
    type: "text",
  },
  {
    font: "cmr10",
    left: 17547621.375,
    text: "ho",
    top: 4128768,
    type: "text",
  },
  {
    font: "cmr10",
    left: 18221187.625,
    text: "w",
    top: 4128768,
    type: "text",
  },
  {
    font: "cmr10",
    left: 18951847.5,
    text: "to",
    top: 4128768,
    type: "text",
  },
  {
    font: "cmr10",
    left: 19791734.875,
    text: "use",
    top: 4128768,
    type: "text",
  },
  {
    font: "cmr10",
    left: 20962943.5,
    text: "the",
    top: 4128768,
    type: "text",
  },
  {
    font: "cmtt10",
    left: 22130512.5,
    text: "xcolor",
    top: 4128768,
    type: "text",
  },
  {
    font: "cmr10",
    left: 24452223,
    text: "pac",
    top: 4128768,
    type: "text",
  },
  {
    font: "cmr10",
    left: 25417061.125,
    text: "k",
    top: 4128768,
    type: "text",
  },
  {
    font: "cmr10",
    left: 25726538.375,
    text: "age",
    top: 4128768,
    type: "text",
  },
  {
    font: "cmr10",
    left: 4063232,
    text: "to",
    top: 4915200,
    type: "text",
  },
  {
    font: "cmr10",
    left: 4864229.375,
    text: "c",
    top: 4915200,
    type: "text",
  },
  {
    font: "cmr10",
    left: 5137296.25,
    text: "hange",
    top: 4915200,
    type: "text",
  },
  {
    font: "cmr10",
    left: 7030563.625,
    text: "the",
    top: 4915200,
    type: "text",
  },
  {
    font: "cmr10",
    left: 8159241.625,
    text: "colour",
    top: 4915200,
    type: "text",
  },
  {
    font: "cmr10",
    left: 10127147.75,
    text: "of",
    top: 4915200,
    type: "text",
  },
  {
    font: "cmr10",
    left: 10873532,
    text: "elemen",
    top: 4915200,
    type: "text",
  },
  {
    font: "cmr10",
    left: 12821412.625,
    text: "ts",
    top: 4915200,
    type: "text",
  },
  {
    font: "cmr10",
    left: 13553232.5,
    text: "in",
    top: 4915200,
    type: "text",
  },
  {
    font: "cmr10",
    left: 14317820.5,
    text: "L",
    top: 4915200,
    type: "text",
  },
  {
    font: "cmr7",
    left: 14491491.75,
    text: "A",
    top: 4780853,
    type: "text",
  },
  {
    font: "cmr10",
    left: 14780038.75,
    text: "T",
    top: 4915200,
    type: "text",
  },
  {
    font: "cmr10",
    left: 15144105.625,
    text: "E",
    top: 5056284,
    type: "text",
  },
  {
    font: "cmr10",
    left: 15508195.625,
    text: "X.",
    top: 4915200,
    type: "text",
  },
  {
    color: "#0000ff",
    type: "color",
  },
  {
    color: "#0000ff",
    type: "color",
  },
  {
    font: "tcrm1000",
    left: 5046354,
    text: "",
    top: 6356992,
    type: "text",
  },
  {
    color: "#0000ff",
    type: "color",
  },
  {
    font: "cmr10",
    left: 5701634,
    text: "First",
    top: 6356992,
    type: "text",
  },
  {
    font: "cmr10",
    left: 7299988.875,
    text: "item",
    top: 6356992,
    type: "text",
  },
  {
    color: "#0000ff",
    type: "color",
  },
  {
    font: "tcrm1000",
    left: 5046354,
    text: "",
    top: 7667712,
    type: "text",
  },
  {
    color: "#0000ff",
    type: "color",
  },
  {
    font: "cmr10",
    left: 5701634,
    text: "Second",
    top: 7667712,
    type: "text",
  },
  {
    font: "cmr10",
    left: 7922582,
    text: "item",
    top: 7667712,
    type: "text",
  },
  {
    color: "black",
    type: "color",
  },
  {
    color: "#ff0000",
    type: "color",
  },
  {
    height: 93233,
    left: 4063232,
    top: 9109504,
    type: "rect",
    width: 22609920,
  },
  {
    color: "black",
    type: "color",
  },
  {
    color: "black",
    type: "color",
  },
  {
    font: "cmr10",
    left: 15204352,
    text: "1",
    top: 41484288,
    type: "text",
  },
  {
    color: "black",
    type: "color",
  },
]
`;

snapshot[`parse() 2`] = `
[
  {
    comment: " TeX output 2017.07.23:1405",
    denominator: 473628672,
    magnification: 1000,
    numerator: 25400000,
    type: "info",
    version: 2,
  },
  {
    font: "cmr17",
    left: 3485199,
    text: "A",
    top: 7503856,
    type: "text",
  },
  {
    font: "cmr17",
    left: 4837893.925843239,
    text: "L",
    top: 7503856,
    type: "text",
  },
  {
    font: "cmr12",
    left: 5171935.543190956,
    text: "A",
    top: 7220124,
    type: "text",
  },
  {
    font: "cmr17",
    left: 5677644.543190956,
    text: "T",
    top: 7503856,
    type: "text",
  },
  {
    font: "cmr17",
    left: 6377217.301024437,
    text: "E",
    top: 7796421,
    type: "text",
  },
  {
    font: "cmr17",
    left: 7075250.301024437,
    text: "X",
    top: 7503856,
    type: "text",
  },
  {
    font: "cmr17",
    left: 8204615.226867676,
    text: "2",
    top: 7503856,
    type: "text",
  },
  {
    font: "cmmi12",
    left: 8827976.831970215,
    text: '"',
    top: 7707738,
    type: "text",
  },
  {
    font: "cmr17",
    left: 9893882.925843239,
    text: "T",
    top: 7503856,
    type: "text",
  },
  {
    font: "cmr17",
    left: 10694770.68367672,
    text: "emplate",
    top: 7503856,
    type: "text",
  },
  {
    font: "cmr17",
    left: 15397623.325587273,
    text: "to",
    top: 7503856,
    type: "text",
  },
  {
    font: "cmr17",
    left: 16912655.45942688,
    text: "test",
    top: 7503856,
    type: "text",
  },
  {
    font: "cmr17",
    left: 19326460.916314125,
    text: "the",
    top: 7503856,
    type: "text",
  },
  {
    font: "cmr17",
    left: 21464855.951506615,
    text: "dvi",
    top: 7503856,
    type: "text",
  },
  {
    font: "cmr17",
    left: 23567745.39354515,
    text: "library",
    top: 7503856,
    type: "text",
  },
  {
    font: "cmr12",
    left: 12486446,
    text: "Ric",
    top: 9838564,
    type: "text",
  },
  {
    font: "cmr12",
    left: 13808184.528894424,
    text: "hard",
    top: 9838564,
    type: "text",
  },
  {
    font: "cmr12",
    left: 15964294.74556923,
    text: "Do",
    top: 9838564,
    type: "text",
  },
  {
    font: "cmr12",
    left: 17157692.35062027,
    text: "dd",
    top: 9838564,
    type: "text",
  },
  {
    font: "cmr12",
    left: 11190215,
    text: "A",
    top: 11018212,
    type: "text",
  },
  {
    font: "cmr12",
    left: 12190939.006391525,
    text: "citizen",
    top: 11018212,
    type: "text",
  },
  {
    font: "cmr12",
    left: 15117088.896680832,
    text: "of",
    top: 11018212,
    type: "text",
  },
  {
    font: "cmr12",
    left: 16169475.58117485,
    text: "no",
    top: 11018212,
    type: "text",
  },
  {
    font: "cmr12",
    left: 17119191.16775322,
    text: "where",
    top: 11018212,
    type: "text",
  },
  {
    font: "cmr12",
    left: 13153812,
    text: "The",
    top: 12197860,
    type: "text",
  },
  {
    font: "cmr12",
    left: 15053242.32546997,
    text: "W",
    top: 12197860,
    type: "text",
  },
  {
    font: "cmr12",
    left: 15925627.22306633,
    text: "orld",
    top: 12197860,
    type: "text",
  },
  {
    height: 26213,
    left: 13153812,
    top: 12302712,
    type: "rect",
    width: 4363224,
  },
  {
    font: "cmr12",
    left: 12646789,
    text: "July",
    top: 14042693,
    type: "text",
  },
  {
    font: "cmr12",
    left: 14687230.365701675,
    text: "23,",
    top: 14042693,
    type: "text",
  },
  {
    font: "cmr12",
    left: 16175972.865243912,
    text: "2017",
    top: 14042693,
    type: "text",
  },
  {
    font: "cmbx10",
    left: 13756622,
    text: "Abstract",
    top: 16520879,
    type: "text",
  },
  {
    font: "cmr10",
    left: 5557435,
    text: "This",
    top: 17431378,
    type: "text",
  },
  {
    font: "cmr10",
    left: 7196002.019373894,
    text: "is",
    top: 17431378,
    type: "text",
  },
  {
    font: "cmr10",
    left: 7917608.766179085,
    text: "the",
    top: 17431378,
    type: "text",
  },
  {
    font: "cmr10",
    left: 9153510.86340046,
    text: "pap",
    top: 17431378,
    type: "text",
  },
  {
    font: "cmr10",
    left: 10329612.609927177,
    text: "er's",
    top: 17431378,
    type: "text",
  },
  {
    font: "cmr10",
    left: 11651230.598884583,
    text: "abstract",
    top: 17431378,
    type: "text",
  },
  {
    font: "cmr10",
    left: 14447959.632979393,
    text: ".",
    top: 17431378,
    type: "text",
  },
  {
    font: "cmr10",
    left: 14766901.852423668,
    text: ".",
    top: 17431378,
    type: "text",
  },
  {
    font: "cmr10",
    left: 15085844.071867943,
    text: ".",
    top: 17431378,
    type: "text",
  },
  {
    font: "cmbx12",
    left: 2555904,
    text: "1",
    top: 20004014,
    type: "text",
  },
  {
    font: "cmbx12",
    left: 4466932,
    text: "In",
    top: 20004014,
    type: "text",
  },
  {
    font: "cmbx12",
    left: 5613812.5164814,
    text: "tro",
    top: 20004014,
    type: "text",
  },
  {
    font: "cmbx12",
    left: 7302020.439722061,
    text: "duction",
    top: 20004014,
    type: "text",
  },
  {
    font: "cmr12",
    left: 2555904,
    text: "This",
    top: 21733073,
    type: "text",
  },
  {
    font: "cmr12",
    left: 4314162,
    text: "do",
    top: 21733073,
    type: "text",
  },
  {
    font: "cmr12",
    left: 5148372,
    text: "cumen",
    top: 21733073,
    type: "text",
  },
  {
    font: "cmr12",
    left: 7308762,
    text: "t",
    top: 21733073,
    type: "text",
  },
  {
    font: "cmr12",
    left: 7864902,
    text: "exists",
    top: 21733073,
    type: "text",
  },
  {
    font: "cmr12",
    left: 9991068,
    text: "solely",
    top: 21733073,
    type: "text",
  },
  {
    font: "cmr12",
    left: 12112956,
    text: "for",
    top: 21733073,
    type: "text",
  },
  {
    font: "cmr12",
    left: 13289406,
    text: "the",
    top: 21733073,
    type: "text",
  },
  {
    font: "cmr12",
    left: 14615586,
    text: "purp",
    top: 21733073,
    type: "text",
  },
  {
    font: "cmr12",
    left: 16219836,
    text: "ose",
    top: 21733073,
    type: "text",
  },
  {
    font: "cmr12",
    left: 17507514,
    text: "of",
    top: 21733073,
    type: "text",
  },
  {
    font: "cmr12",
    left: 18384504,
    text: "testing",
    top: 21733073,
    type: "text",
  },
  {
    font: "cmr12",
    left: 20912802,
    text: "the",
    top: 21733073,
    type: "text",
  },
  {
    font: "cmr12",
    left: 22238982,
    text: "dvi",
    top: 21733073,
    type: "text",
  },
  {
    font: "cmr12",
    left: 23543772,
    text: "library",
    top: 21733073,
    type: "text",
  },
  {
    font: "cmbx12",
    left: 2555904,
    text: "Outline",
    top: 23729538,
    type: "text",
  },
  {
    font: "cmr12",
    left: 6212713.5,
    text: "The",
    top: 23729538,
    type: "text",
  },
  {
    font: "cmr12",
    left: 7867136.5,
    text: "remainder",
    top: 23729538,
    type: "text",
  },
  {
    font: "cmr12",
    left: 11574998.5,
    text: "of",
    top: 23729538,
    type: "text",
  },
  {
    font: "cmr12",
    left: 12523551.5,
    text: "this",
    top: 23729538,
    type: "text",
  },
  {
    font: "cmr12",
    left: 14096691.5,
    text: "article",
    top: 23729538,
    type: "text",
  },
  {
    font: "cmr12",
    left: 16521154.5,
    text: "is",
    top: 23729538,
    type: "text",
  },
  {
    font: "cmr12",
    left: 17367034.5,
    text: "organized",
    top: 23729538,
    type: "text",
  },
  {
    font: "cmr12",
    left: 20903777.5,
    text: "as",
    top: 23729538,
    type: "text",
  },
  {
    font: "cmr12",
    left: 21920777.5,
    text: "follo",
    top: 23729538,
    type: "text",
  },
  {
    font: "cmr12",
    left: 23332517.5,
    text: "ws.",
    top: 23729538,
    type: "text",
  },
  {
    font: "cmr12",
    left: 24963223.5,
    text: "Section",
    top: 23729538,
    type: "text",
  },
  {
    font: "cmr12",
    left: 27729925.5,
    text: "2",
    top: 23729538,
    type: "text",
  },
  {
    font: "cmr12",
    left: 2555904,
    text: "giv",
    top: 24679810,
    type: "text",
  },
  {
    font: "cmr12",
    left: 3539844,
    text: "es",
    top: 24679810,
    type: "text",
  },
  {
    font: "cmr12",
    left: 4463737,
    text: "accoun",
    top: 24679810,
    type: "text",
  },
  {
    font: "cmr12",
    left: 6752467,
    text: "t",
    top: 24679810,
    type: "text",
  },
  {
    font: "cmr12",
    left: 7329843,
    text: "of",
    top: 24679810,
    type: "text",
  },
  {
    font: "cmr12",
    left: 8228068,
    text: "previous",
    top: 24679810,
    type: "text",
  },
  {
    font: "cmr12",
    left: 11312352,
    text: "w",
    top: 24679810,
    type: "text",
  },
  {
    font: "cmr12",
    left: 11847102,
    text: "ork.",
    top: 24679810,
    type: "text",
  },
  {
    font: "cmr12",
    left: 13557838,
    text: "Our",
    top: 24679810,
    type: "text",
  },
  {
    font: "cmr12",
    left: 15161933,
    text: "new",
    top: 24679810,
    type: "text",
  },
  {
    font: "cmr12",
    left: 16766028,
    text: "and",
    top: 24679810,
    type: "text",
  },
  {
    font: "cmr12",
    left: 18284564,
    text: "exciting",
    top: 24679810,
    type: "text",
  },
  {
    font: "cmr12",
    left: 21193449,
    text: "results",
    top: 24679810,
    type: "text",
  },
  {
    font: "cmr12",
    left: 23661701,
    text: "are",
    top: 24679810,
    type: "text",
  },
  {
    font: "cmr12",
    left: 24966336,
    text: "describ",
    top: 24679810,
    type: "text",
  },
  {
    font: "cmr12",
    left: 27344904,
    text: "ed",
    top: 24679810,
    type: "text",
  },
  {
    font: "cmr12",
    left: 2555904,
    text: "in",
    top: 25630082,
    type: "text",
  },
  {
    font: "cmr12",
    left: 3454284,
    text: "Section",
    top: 25630082,
    type: "text",
  },
  {
    font: "cmr12",
    left: 6149424,
    text: "3.",
    top: 25630082,
    type: "text",
  },
  {
    font: "cmr12",
    left: 7090584,
    text: "Finally",
    top: 25630082,
    type: "text",
  },
  {
    font: "cmr12",
    left: 9389873.25,
    text: ",",
    top: 25630082,
    type: "text",
  },
  {
    font: "cmr12",
    left: 9860453.25,
    text: "Section",
    top: 25630082,
    type: "text",
  },
  {
    font: "cmr12",
    left: 12555593.25,
    text: "4",
    top: 25630082,
    type: "text",
  },
  {
    font: "cmr12",
    left: 13197293.25,
    text: "giv",
    top: 25630082,
    type: "text",
  },
  {
    font: "cmr12",
    left: 14181233.25,
    text: "es",
    top: 25630082,
    type: "text",
  },
  {
    font: "cmr12",
    left: 15083891.25,
    text: "the",
    top: 25630082,
    type: "text",
  },
  {
    font: "cmr12",
    left: 16410071.25,
    text: "conclusions.",
    top: 25630082,
    type: "text",
  },
  {
    font: "cmbx12",
    left: 2555904,
    text: "2",
    top: 28202719,
    type: "text",
  },
  {
    font: "cmbx12",
    left: 4466932,
    text: "Previous",
    top: 28202719,
    type: "text",
  },
  {
    font: "cmbx12",
    left: 9737209.337694168,
    text: "w",
    top: 28202719,
    type: "text",
  },
  {
    font: "cmbx12",
    left: 10621944.712694168,
    text: "ork",
    top: 28202719,
    type: "text",
  },
  {
    font: "cmr12",
    left: 2555904,
    text: "A",
    top: 29931778,
    type: "text",
  },
  {
    font: "cmr12",
    left: 3389841.75,
    text: "made",
    top: 29931778,
    type: "text",
  },
  {
    font: "cmr12",
    left: 5443281.75,
    text: "up",
    top: 29931778,
    type: "text",
  },
  {
    font: "cmr12",
    left: 6555561.75,
    text: "guide",
    top: 29931778,
    type: "text",
  },
  {
    font: "cmr12",
    left: 8609001.75,
    text: "to",
    top: 29931778,
    type: "text",
  },
  {
    font: "cmr12",
    left: 9550161.75,
    text: "the",
    top: 29931778,
    type: "text",
  },
  {
    font: "cmr12",
    left: 10876341.75,
    text: "dvi",
    top: 29931778,
    type: "text",
  },
  {
    font: "cmr12",
    left: 12181131.75,
    text: "format",
    top: 29931778,
    type: "text",
  },
  {
    font: "cmr12",
    left: 14683761.75,
    text: "b",
    top: 29931778,
    type: "text",
  },
  {
    font: "cmr12",
    left: 15090171.75,
    text: "y",
    top: 29931778,
    type: "text",
  },
  {
    font: "cmr12",
    left: 15753261.75,
    text: "Anon",
    top: 29931778,
    type: "text",
  },
  {
    font: "cmr12",
    left: 17827819.5,
    text: "[",
    top: 29931778,
    type: "text",
  },
  {
    font: "cmr12",
    left: 18041719.5,
    text: "1",
    top: 29931778,
    type: "text",
  },
  {
    font: "cmr12",
    left: 18426739.5,
    text: "].",
    top: 29931778,
    type: "text",
  },
  {
    font: "cmr12",
    left: 2555904,
    text: "A",
    top: 30882050,
    type: "text",
  },
  {
    font: "cmr12",
    left: 3389841.75,
    text: "real",
    top: 30882050,
    type: "text",
  },
  {
    font: "cmr12",
    left: 4887141.75,
    text: "guide",
    top: 30882050,
    type: "text",
  },
  {
    font: "cmr12",
    left: 6940581.75,
    text: "to",
    top: 30882050,
    type: "text",
  },
  {
    font: "cmr12",
    left: 7881741.75,
    text: "the",
    top: 30882050,
    type: "text",
  },
  {
    font: "cmr12",
    left: 9207921.75,
    text: "dvi",
    top: 30882050,
    type: "text",
  },
  {
    font: "cmr12",
    left: 10512711.75,
    text: "format",
    top: 30882050,
    type: "text",
  },
  {
    font: "cmr12",
    left: 13015341.75,
    text: "[",
    top: 30882050,
    type: "text",
  },
  {
    font: "cmr12",
    left: 13229241.75,
    text: "2",
    top: 30882050,
    type: "text",
  },
  {
    font: "cmr12",
    left: 13614261.75,
    text: "].",
    top: 30882050,
    type: "text",
  },
  {
    font: "cmbx12",
    left: 2555904,
    text: "3",
    top: 33454686,
    type: "text",
  },
  {
    font: "cmbx12",
    left: 4466932,
    text: "Results",
    top: 33454686,
    type: "text",
  },
  {
    font: "cmr12",
    left: 2555904,
    text: "In",
    top: 35183745,
    type: "text",
  },
  {
    font: "cmr12",
    left: 3518181.75,
    text: "this",
    top: 35183745,
    type: "text",
  },
  {
    font: "cmr12",
    left: 5019759.75,
    text: "section",
    top: 35183745,
    type: "text",
  },
  {
    font: "cmr12",
    left: 7590837.75,
    text: "w",
    top: 35183745,
    type: "text",
  },
  {
    font: "cmr12",
    left: 8125587.75,
    text: "e",
    top: 35183745,
    type: "text",
  },
  {
    font: "cmr12",
    left: 8724507.75,
    text: "describ",
    top: 35183745,
    type: "text",
  },
  {
    font: "cmr12",
    left: 11103075.75,
    text: "e",
    top: 35183745,
    type: "text",
  },
  {
    font: "cmr12",
    left: 11701995.75,
    text: "the",
    top: 35183745,
    type: "text",
  },
  {
    font: "cmr12",
    left: 13028175.75,
    text: "results.",
    top: 35183745,
    type: "text",
  },
  {
    font: "cmbx12",
    left: 2555904,
    text: "4",
    top: 37756381,
    type: "text",
  },
  {
    font: "cmbx12",
    left: 4466932,
    text: "Conclusions",
    top: 37756381,
    type: "text",
  },
  {
    font: "cmr12",
    left: 2555904,
    text: "W",
    top: 39485440,
    type: "text",
  },
  {
    font: "cmr12",
    left: 3282891.75,
    text: "e",
    top: 39485440,
    type: "text",
  },
  {
    font: "cmr12",
    left: 3881811.75,
    text: "w",
    top: 39485440,
    type: "text",
  },
  {
    font: "cmr12",
    left: 4416561.75,
    text: "ork",
    top: 39485440,
    type: "text",
  },
  {
    font: "cmr12",
    left: 5486061.75,
    text: "ed",
    top: 39485440,
    type: "text",
  },
  {
    font: "cmr12",
    left: 6512781.75,
    text: "hard,",
    top: 39485440,
    type: "text",
  },
  {
    font: "cmr12",
    left: 8523441.75,
    text: "and",
    top: 39485440,
    type: "text",
  },
  {
    font: "cmr12",
    left: 10020741.75,
    text: "ac",
    top: 39485440,
    type: "text",
  },
  {
    font: "cmr12",
    left: 10726611.75,
    text: "hiev",
    top: 39485440,
    type: "text",
  },
  {
    font: "cmr12",
    left: 12095571.75,
    text: "ed",
    top: 39485440,
    type: "text",
  },
  {
    font: "cmr12",
    left: 13122291.75,
    text: "v",
    top: 39485440,
    type: "text",
  },
  {
    font: "cmr12",
    left: 13507311.75,
    text: "ery",
    top: 39485440,
    type: "text",
  },
  {
    font: "cmr12",
    left: 14812101.75,
    text: "little.",
    top: 39485440,
    type: "text",
  },
  {
    font: "cmr12",
    left: 15142914,
    text: "1",
    top: 41451520,
    type: "text",
  },
  {
    font: "cmbx12",
    left: 2555904,
    text: "References",
    top: 4325376,
    type: "text",
  },
  {
    font: "cmr12",
    left: 2555904,
    text: "[1]",
    top: 6054435,
    type: "text",
  },
  {
    font: "cmr12",
    left: 3753744,
    text: "A.",
    top: 6054435,
    type: "text",
  },
  {
    font: "cmr12",
    left: 4824781.75,
    text: "N.",
    top: 6054435,
    type: "text",
  },
  {
    font: "cmr12",
    left: 5895820.5,
    text: "Other.",
    top: 6054435,
    type: "text",
  },
  {
    font: "cmr12",
    left: 8488121.5,
    text: "Dvi",
    top: 6054435,
    type: "text",
  },
  {
    font: "cmr12",
    left: 9976401.75,
    text: "for",
    top: 6054435,
    type: "text",
  },
  {
    font: "cmr12",
    left: 11176051.75,
    text: "dummies.",
    top: 6054435,
    type: "text",
  },
  {
    font: "cmr12",
    left: 14799350.75,
    text: "man",
    top: 6054435,
    type: "text",
  },
  {
    font: "cmr12",
    left: 16232480.75,
    text: "uscript,",
    top: 6054435,
    type: "text",
  },
  {
    font: "cmr12",
    left: 19046459.75,
    text: "San",
    top: 6054435,
    type: "text",
  },
  {
    font: "cmr12",
    left: 20265689.75,
    text: "ta's",
    top: 6054435,
    type: "text",
  },
  {
    font: "cmr12",
    left: 21747687.75,
    text: "Grotto,",
    top: 6054435,
    type: "text",
  },
  {
    font: "cmr12",
    left: 24519888.75,
    text: "The",
    top: 6054435,
    type: "text",
  },
  {
    font: "cmr12",
    left: 26125948.75,
    text: "North",
    top: 6054435,
    type: "text",
  },
  {
    font: "cmr12",
    left: 3753744,
    text: "P",
    top: 7004707,
    type: "text",
  },
  {
    font: "cmr12",
    left: 4256273.25,
    text: "ole,",
    top: 7004707,
    type: "text",
  },
  {
    font: "cmr12",
    left: 5668013.25,
    text: "2003.",
    top: 7004707,
    type: "text",
  },
  {
    font: "cmr12",
    left: 2555904,
    text: "[2]",
    top: 8610339,
    type: "text",
  },
  {
    font: "cmr12",
    left: 3753744,
    text: "A.",
    top: 8610339,
    type: "text",
  },
  {
    font: "cmr12",
    left: 5615081.75,
    text: "Sno",
    top: 8610339,
    type: "text",
  },
  {
    font: "cmr12",
    left: 6834311.75,
    text: "wden.",
    top: 8610339,
    type: "text",
  },
  {
    font: "cmr12",
    left: 11567811.75,
    text: "Dvi",
    top: 8610339,
    type: "text",
  },
  {
    font: "cmr12",
    left: 13846391,
    text: "sp",
    top: 8610339,
    type: "text",
  },
  {
    font: "cmr12",
    left: 14599319,
    text: "ecication.",
    top: 8610339,
    type: "text",
  },
  {
    font: "cmr12",
    left: 20958459,
    text: "w",
    top: 8610339,
    type: "text",
  },
  {
    font: "cmr12",
    left: 21493209,
    text: "eb",
    top: 8610339,
    type: "text",
  },
  {
    font: "cmr12",
    left: 23333430,
    text: "page,",
    top: 8610339,
    type: "text",
  },
  {
    font: "cmr12",
    left: 26360965,
    text: "2007.",
    top: 8610339,
    type: "text",
  },
  {
    font: "cmtt12",
    left: 3753744,
    text: "https://web.archive.org/web/20070403030353/",
    top: 9560611,
    type: "text",
  },
  {
    font: "cmtt12",
    left: 3753744,
    text: "http://www.math.umd.edu/",
    top: 10510883,
    type: "text",
  },
  {
    font: "cmtt12",
    left: 13870869,
    text: "asnowden/comp-cont/dvi.html",
    top: 10510883,
    type: "text",
  },
  {
    font: "cmr12",
    left: 24797364,
    text: ".",
    top: 10510883,
    type: "text",
  },
  {
    font: "cmr12",
    left: 15142914,
    text: "2",
    top: 41451520,
    type: "text",
  },
]
`;
